apps/console/src/features/identity-providers/components/settings/attribute-management/attribute-selection-modal.tsx

```diff
@@ -0,0 +1,164 @@
+): ReactElement => {
+
+    const {
+        attributeList,
+        show,
+        onClose,
+        onSave
+    } = props;
+
+    /**
+     * The claims that user picks. All the selecting ones will be
+     * pushed to this piece of state.
+     */
+    const [ claimsToBeAdded, setClaimsToBeAdded ] = useState<IdentityProviderCommonClaimMappingInterface[]>([]);
+    /**
+     * Since this modal is a intermediate step in adding attributes
+     * we can't read the records always from from {@link attributeList}
+     * we need to internally remove selected ones from the dropdown.
+     */
+    const [ copyOfAttributes, setCopyOfAttributes ] = useState<IdentityProviderClaimInterface[]>([]);
+    /**
+     * This has the state to tell whether {@link claimsToBeAdded} is empty
+     * or not. We need this to control the {@link Transition} state.
+     */
+    const [ showPlaceholder, setShowPlaceholder ] = useState<boolean>(false);
+
+    useEffect(() => {
+        // Clone it first. We don't want to mutate the original list.
+        setCopyOfAttributes([ ...attributeList ]);
+    }, []);
+
+    useEffect(() => {
+        setShowPlaceholder(claimsToBeAdded.length === 0);
+    }, [ claimsToBeAdded ]);
+
+    const onAttributeMappingAdd = (mapping: IdentityProviderCommonClaimMappingInterface): void => {
+        const newClaimsToBeAdded = [ ...claimsToBeAdded, mapping ];
+        const idsToHide = newClaimsToBeAdded.reduce((acc, value) => acc.add(value.claim.id), new Set<string>());
+        // Records to be added.
+        setClaimsToBeAdded(newClaimsToBeAdded);
+        //  Remove the added attribute from the attribute list.
+        setCopyOfAttributes([ ...copyOfAttributes.filter(({ id }) => !idsToHide.has(id)) ]);
+    };
+
+    const onMappingDeleted = (mapping: IdentityProviderCommonClaimMappingInterface): void => {
+        const filtered = claimsToBeAdded.filter((m) => (
+            m.claim.id !== mapping.claim.id
+        ));
+        setClaimsToBeAdded([ ...filtered ]);
+        setCopyOfAttributes([ ...copyOfAttributes, mapping.claim ]);
+    };
+
+    const onMappingEdited = (
+        oldMapping: IdentityProviderCommonClaimMappingInterface,
+        newMapping: IdentityProviderCommonClaimMappingInterface
+    ) => {
+        // If user changed the local mapping of this. Then first
+        // go and remove it from {@link claimsToBeAdded} array.
+        setClaimsToBeAdded([
+            ...claimsToBeAdded
+                .filter(({ claim }) => (claim.id !== oldMapping.claim.id)),
+            newMapping
+        ]);
+    };
+
+    const _placeholder = (): ReactElement => {
+        return (
+            <EmptyPlaceholder
+                title={ "You haven't selected any attributes" }
+                subtitle={
+                    [
+                        <p key={ "empty-attributes-mapping" }>
+                            Map attributes and click <strong>Add Attribute Mapping</strong> to get started.
+                        </p>
```

Comment:
Please move to localisation files.

Reply:
Tracking an issue for this. I will address them with https://github.com/wso2-enterprise/asgardeo-product/issues/5593